### PR TITLE
Changes to make advanced options work

### PR DESCRIPTION
**Pull request by puneet-pdx**
Related to issue: # https://devtopia.esri.com/runtime/kotlin/issues/4644

### Description:

Make sure selections made by the user in Advanced options work when a trace is executed

### Summary of changes:

- Rename `currentTraceGraphics` to `currentTraceGeometryResultGraphics` to reflect the usage of the parameter, also don't add starting point graphic to it
- Hook up the UI to update the `TraceState` with the user selected values for name, trace graphic colors and zoom to result 
- add logic to determine the Name for the trace, update graphic colors and zoom to result

### Pre-merge Checklist

- a [vTest](https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/) Job for this PR has been run
  - [x] link: https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/117/
- Unit and/or integration tests have been added to exercise this PR's logic, and the tests are passing:
  - [ ] Yes
  - [ ] No

### CODE REVIEW

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceOptionsScreen.kt [hunk 1]
```diff
@@ -328,15 +345,23 @@ internal fun AdvancedOptions(
...
+                Row(
+                    modifier = modifier
+                        .fillMaxWidth()
+                        .height(80.dp)
+                        .padding(10.dp),
+                    verticalAlignment = Alignment.CenterVertically,
+                    horizontalArrangement = Arrangement.SpaceBetween
+                ) {
+                    var text by rememberSaveable { mutableStateOf(defaultTraceName) }
                     TextField(
```
Comment (shubham7109) [inline toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceOptionsScreen.kt:357]: Probably subjective, but I find the `OutlinedTextField` is a nicer looking text field for user inputs: 
```suggestion
                    OutlinedTextField(
```

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceOptionsScreen.kt [hunk 2]
```diff
@@ -483,7 +507,7 @@ private fun AdvancedOptionsRow(name: String, modifier: Modifier = Modifier, trai
 @Preview
```
Comment (shubham7109) [inline toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceOptionsScreen.kt:507]:
```suggestion
@Preview(showBackground = true)
```

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceOptionsScreen.kt [hunk 3]
```diff
@@ -483,7 +507,7 @@ private fun AdvancedOptionsRow(name: String, modifier: Modifier = Modifier, trai
...
 @Composable
 private fun AdvancedOptionsPreview() {
-    AdvancedOptions()
+    AdvancedOptions(defaultTraceName = "", selectedColor = Color.Green, zoomToResult = false)
 }
 
 @Preview
```
Comment (shubham7109) [inline toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceOptionsScreen.kt:513]:
```suggestion
@Preview(showBackground = true)
```

**Review by shubham7109 — APPROVED** [review]
Looks good, just a few minor comments. Happy to approve.

**Comment by puneet-pdx** [comment]
Thanks @shubham7109 ! Merging..

**Pull request merged**